src/Message/CIMGetProfileResponse.php

```diff
@@ -28,9 +28,11 @@ public function getMatchingPaymentProfileId($last4)
 
         foreach ($this->data['profile']['paymentProfiles'] as $paymentProfile) {
             // For every payment  profile check if the last4 matches the last4 of the card in request.
-            $cardLast4 = substr($paymentProfile['payment']['creditCard']['cardNumber'], -4);
-            if ($last4 == $cardLast4) {
-                return (string)$paymentProfile['customerPaymentProfileId'];
+            if (isset($paymentProfile['payment']['creditCard']['cardNumber'])) {
+                $cardLast4 = substr($paymentProfile['payment']['creditCard']['cardNumber'], -4);
+                if ($last4 == $cardLast4) {
+                    return (string)$paymentProfile['customerPaymentProfileId'];
+                }
```

Comment:
Can you add a TODO to investigate and fixed later

Reply:
done